File: demos/clinical-trial-portal-go/hash.go
package main

import (
	"crypto/sha256"
	"encoding/hex"
)

// sha256hex — SHA-256 of input as lowercase hex. Same primitive as render 1/2's
// lib/hash.ts (node:crypto createHash("sha256")); SHA-256 is bit-identical across
// languages, so a Go-produced hash equals the JS one for the same canonical bytes.
func sha256hex(input string) string {
	sum := sha256.Sum256([]byte(input))
	return hex.EncodeToString(sum[:])
}

File: demos/clinical-trial-portal-mongo/lib/hash.mjs
/**
 * lib/hash.mjs — SHA-256 + random tokens, ported from render 2
 * (demos/clinical-trial-portal-next/lib/hash.ts). Same primitive
 * (node:crypto), synchronous sha256hex so it can run inside the
 * transaction body.
 */
import { createHash, randomBytes } from "node:crypto";

/** SHA-256 of `input` as lowercase hex. Synchronous — safe inside a transaction. */
export function sha256hex(input) {
  return createHash("sha256").update(input).digest("hex");
}

/** Cryptographically-random token as lowercase hex of length `bytes * 2`. */
export function randomToken(bytes = 32) {
  return randomBytes(bytes).toString("hex");
}
